### PR TITLE
Closes #155: add inventory participants

### LINKED ISSUE #155: Product Inventory: Add Participant
Show a modal to add new participant to inventory as a link/button/directive from the old modal/style for inventories. This modal already exists, we just want to show it using a link/directive.

### PR DESCRIPTION
**Pull request by bithavoc**

### CODE REVIEW

**Comment by MakotoTheKnight** [comment]
Mind rebasing this branch against staging?

File: app/assets/javascripts/client/inventories/InventoryInvitation.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+(function() {
+  'use strict';
+
+  angular.module('PDRClient').factory('InventoryInvitation', ['$resource', 'UrlService', function($resource, UrlService) {
```
Comment (MakotoTheKnight) [inline app/assets/javascripts/client/inventories/InventoryInvitation.js]: Style check:  this should be using the same style we've used for all factories, controllers, and directives, in that the function for the factory and its injectables are broken out.

File: app/assets/javascripts/client/inventories/InviteInventoryUserCtrl.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+(function() {
+  'use strict';
+
+  angular.module('PDRClient').controller('InviteInventoryUserCtrl', InviteInventoryUserCtrl);
+
+  InviteInventoryUserCtrl.$inject = ['$scope', 'InventoryInvitation'];
+  function InviteInventoryUserCtrl($scope, InventoryInvitation) {
+    var vm = this;
+    vm.sendInvitation = function(invitation) {
+      $scope.alerts  = [];
```
Comment (MakotoTheKnight) [inline app/assets/javascripts/client/inventories/InviteInventoryUserCtrl.js]: This is...strange.  We're defining a field in `$scope` but we're using `vm`?  Also, it's done inside of a method which may never be invoked.  If we attempt to use `alerts` for anything it's undefined on the scope until we invoke `sendInvitation`.

File: app/assets/javascripts/client/inventories/InviteInventoryUserCtrl.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      $scope.addAlert = function(message) {
```
Comment (MakotoTheKnight) [inline app/assets/javascripts/client/inventories/InviteInventoryUserCtrl.js]: For this, we really should be using `vm` as opposed to nesting this inside of this function.

File: app/assets/javascripts/client/inventories/InviteInventoryUserCtrl.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        $scope.alerts.push({type: 'danger', msg: message});
+      };
+
+      $scope.closeAlert = function(index) {
+        $scope.alerts.splice(index, 1);
+      };
+      InventoryInvitation.create({ inventory_id: $scope.inventoryId }, invitation).$promise.then(function(response) {
+        var errors = response.data.errors;
```
Comment (MakotoTheKnight) [inline app/assets/javascripts/client/inventories/InviteInventoryUserCtrl.js]: Two things:

 - We're in the context of a fulfilled promise; why would we expect errors?
 - We're not using the variable for anything.

File: app/assets/javascripts/client/inventories/invite_inventory_user.html [hunk 5]
```diff
@@ -0,0 +1,45 @@
+<form  name="formInviteParticipant"  novalidate>
+  <alert ng-repeat="alert in alerts" type="{{alert.type}}" close="closeAlert($index)">{{alert.msg}}</alert>
```
Comment (MakotoTheKnight) [inline app/assets/javascripts/client/inventories/invite_inventory_user.html]: This function should be prefixed with the name defined in the `controllerAs` block of `inviteInventoryUser`.

File: app/assets/javascripts/client/inventories/InviteInventoryUserCtrl.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  'use strict';
+
+  angular.module('PDRClient').controller('InviteInventoryUserCtrl', InviteInventoryUserCtrl);
+
+  InviteInventoryUserCtrl.$inject = ['$scope', 'InventoryInvitation'];
+  function InviteInventoryUserCtrl($scope, InventoryInvitation) {
+    var vm = this;
+    vm.sendInvitation = function(invitation) {
+      vm.alerts  = [];
+      vm.addAlert = function(message) {
```
Comment (MakotoTheKnight) [inline app/assets/javascripts/client/inventories/InviteInventoryUserCtrl.js]: We should pull these variables and functions out of the nested `sendInvitation` function.

Comment (bithavoc) [inline app/assets/javascripts/client/inventories/InviteInventoryUserCtrl.js]: what's the benefit of that?

**Comment by MakotoTheKnight** [comment]
This seems alright to me.  We're going to want a task to capture the number of participants in the list view.

**Comment by MakotoTheKnight** [comment]
The modal created from this should be larger; the size being used isn't large enough to allow for comfortable viewing of the buttons.

**Comment by bithavoc** [comment]
@MakotoTheKnight that's interesting, I thought it'd be a tweak in the CSS from another upcoming PR.

**Comment by MakotoTheKnight** [comment]
@bithavoc:  The original modals passed the `size: 'lg'` property when calling `$modal.open`, which is what gave it the correct dimensions.

**Comment by MakotoTheKnight** [comment]
Go ahead and rebase this against staging and merge it in - we'll be able to use this in #144.

**Comment by bithavoc** [comment]
@MakotoTheKnight which dialog you couldn't properly see the buttons?

**Pull request merged**